ocaml/xapi/storage_migrate.ml

```diff
@@ -542,6 +551,10 @@ let start' ~task ~dbg ~sr ~vdi ~dp ~url ~dest =
     error " Caught exception %s:%s. Performing cleanup." Api_errors.sr_not_attached sr_uuid;
     perform_cleanup_actions !on_fail;
     raise (Api_errors.Server_error(Api_errors.sr_not_attached,[sr_uuid]))
+  | Api_errors.Server_error(code, param) when code = Api_errors.sr_source_space_insufficient ->
+    error " Caught exception %s:%s, performing cleanup." Api_errors.sr_source_space_insufficient (List.hd param);
+    perform_cleanup_actions !on_fail;
+    raise (Api_errors.Server_error(Api_errors.sr_source_space_insufficient, param))
```

Comment:
I think this clause is redundant - the only thing it does is improve the logging in the case of the new error code. I think it would be better to change the `Printexc.to_string` in the generic exception handler clause below and instead call `Api_errors.to_string`

Reply:
Good point, updated.